[#21] 받은 친구 요청 목록 불러오기 기능 구현

msugo1 opened the pull request:
## 📝 컨트롤러 계층 관련

### 📑 FriendController: getFriendRequestList 메소드 추가

- 현재 사용자의 친구 목록을 불러오기 위한 요청을 매핑할 메소드

- 첫 화면을 표시하기 위한 pageNumber `기본값` 설정

- 받아온 친구목록을 `ResponseEntity`를 활용해 본문에 담아서 화면에 반환

## 📝 서비스 계층 관련

### 📑 FriendService: getFriendRequestList 메소드 추가

- 들어온 요청을 바탕으로 Mapper에 전달할 객체 생성

- 반환된 값은 Controller에 전달

Commit:
[#21] 받은 친구 요청 목록 불러오기 기능 구현

컨트롤러 계층 관련

* FriendController: getFriendRequestList 메소드 추가

- 현재 사용자의 친구 목록을 불러오기 위한 요청을 매핑할 메소드

- 첫 화면을 표시하기 위한 pageNumber `기본값` 설정

- 받아온 친구목록을 `ResponseEntity`를 활용해 본문에 담아서 화면에 반환

서비스 계층 관련

* FriendService: getFriendRequestList 메소드 추가

- 들어온 요청을 바탕으로 `FriendListRequest` 객체를 생성해 Mapper에 전달

- 반환된 값은 Controller에 전달

f-lab-dev commented on src/main/java/me/soo/helloworld/controller/FriendController.java:
```diff
@@ -56,4 +56,12 @@ public void unfriendFriend(@CurrentUser String userId, @PathVariable String targ
     }
+
+    @LoginRequired
+    @GetMapping("/friend-requests/from/other-users")
```
`/friend-requests` 만으로도 표현이 가능해보이는데 뒤에 부수적인 내용을 붙여주신 이유가 있으실까요?

msugo1 replied:
해당 URL을 작성할 때는 저렇게 하면 좀 더 명확할 것 같아서 붙였었는데 다시 생각해보니 없어도 URL자체가 말이 되고, 불필요하게 추가된 `/from/other-users`가 URL의 중요한 요소인 간결성을 해치기도 하네요. 제거해 주었습니다.

f-lab-dev commented on src/main/java/me/soo/helloworld/controller/FriendController.java:
```diff
@@ -56,4 +56,12 @@ public void unfriendFriend(@CurrentUser String userId, @PathVariable String targ
+    public ResponseEntity<List<FriendList>> getFriendRequestList(@CurrentUser String userId,
+                                                          @RequestParam(defaultValue = "1") Integer pageNumber) {
+        List<FriendList> friendRequests = friendService.getFriendRequestList(userId, pageNumber);
+        return ResponseEntity.ok(friendRequests);
```
여기는 굳이 `ResponseEntity`를 리턴할 필요가 없어보입니다~

Commit:
[#21] 피드백 반영

FriendController

- URL 불필요한 부분 제거
(/my-friends/friend-requests/from/other-users ->
/my-friends/friend-requests)

- 불필요한 ResponseEntity 리턴제거
